src/lib/OS/SimOS.cc

```diff
@@ -169,72 +169,113 @@ void SimOS::createInitialProcess() {
   std::vector<RegisterFileStructure> regFileStructure =
       arch->getRegisterFileStructures();
 
-  if (executablePath_ != DEFAULT_STR) {
+  // Get the tid for new Process
+  uint64_t tid = nextFreeTID_;
+  nextFreeTID_++;
+
+  if (!instructionBytes.empty()) {
+    // Construct Process from `instructionBytes`
+    processes_.emplace(tid, std::make_shared<Process>(
```

Comment:
Any reason why emplace is used? does an insert not suffice?

Reply:
see https://github.com/UoB-HPC/SimEng/pull/287#discussion_r1115929041